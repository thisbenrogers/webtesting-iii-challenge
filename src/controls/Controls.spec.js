import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

import Controls from './Controls';

describe("<Controls />", () => {

    it('renders <Controls /> correctly', () => {
      const tree = renderer.create(<Controls />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("Changes text when 'Close Gate' button is clicked", () => {
        const mock = jest.fn();
        const { getByText } = render(<Controls toggleLocked={mock} toggleClosed={mock} />);
        fireEvent.click(getByText("Close Gate"));
        expect(mock).toHaveBeenCalled();
    })
});


