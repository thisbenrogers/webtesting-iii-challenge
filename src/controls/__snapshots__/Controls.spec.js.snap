// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> renders <Controls /> correctly 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;
